DragonSlayer.cpp

```diff
@@ -1,28 +1,62 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
+#include "AttackItem.h"
+#include <assert.h>
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string name_, int hp, int armor) : Character(hp, armor, 4), name(name_) 
+{
+    helpfulItems = makeHelpfulItems();
+    defensiveItems = makeDefensiveItems();
+    attackItems = makeAttackItems(0);
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
+    if( hitPoints <= 0 )
+    {
+        std::cout << getName() << " can't attack. " << getName() << " is dead." << std::endl;
+        std::cout << "make another party member use an item to revive them" << std::endl << std::endl;
+        return;
+    }
+        
+    isDefending = false;
+    
```

Comment:
there is no need to modify the `DragonSlayer::attack` function except where indicated.
- [ ] revise